### PR TITLE
Fixes #8: Returning result from closure

### LINKED ISSUE #8: Functions doesn't allow to return value
What I would like to do:
```rust
let result = temp_env::with_var("FOO", "BAR", || {
    let result = do_something_needing_the_env_variable();
    result
});
```
What I have to do now:
```rust
let mut result = Result<MyType, anyhow::Error>;
temp_env::with_var("FOO", "BAR", || {
    result = do_something_needing_the_env_variable();
});
```
Fixing this shouldn't be breaking change, as returning `()` from the closer would have identical effect as it has now.

I drop a PR if I find time to work out the code change, but that may not happen soon. Sorry.

And thanks for very helpful crate.

### PR DESCRIPTION
**Pull request by michalfita**
I've raised #8 after I struggled with testing code that had to return some results, this PR fixes issue reported.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -281,6 +290,31 @@ mod tests {
         assert!(two_not_set_after.is_err(), "`TWO` must not be set.");
     }
 
+    /// Test whether setting multiple variable is returns result.
```
Comment (vmx) [inline src/lib.rs:293]: I think the word `is` here is to much.

**Review by vmx — APPROVED** [review]
Thanks a lot for the improvements, it's really a good change.

I just found a minor thing, see the inline comment.

**Comment by vmx** [comment]
I forgot to mention, that it's great that you've even updated the documentation, thanks!

**Comment by michalfita** [comment]
My `clippy` haven't shown the error the CI reveals. I'll fix this soon.

**Comment by michalfita** [comment]
@vmx Any chance to get this merged and released soon?

**Comment by vmx** [comment]
Sorry, I totally forgot about it, I'll do that now.

**Comment by michalfita** [comment]
Thanks.

**Pull request merged**